### PR TITLE
prov/util,rxm: optimizations

### PR DESCRIPTION
**Pull request by a-ilango**
- prov/util: use a separate flag to determine if we need to create av hash table
- prov/rxm: use ip_av_create_flags
- prov/rxm: don't include FI_DIRECTED_RECV, FI_SOURCE caps if they're not needed

### CODE REVIEW

File: prov/rxm/src/rxm_init.c [hunk 1]
```diff
@@ -166,6 +166,23 @@ static int rxm_init_info(void)
 	return 0;
 }
 
+static void rxm_alter_info(const struct fi_info *hints, struct fi_info *info)
+{
+	struct fi_info *cur;
+
+	if (!hints)
+		return;
```
Comment (shefty) [inline prov/rxm/src/rxm_init.c]: Should these be removed if no hints were provided, since they do affect the fast path?

**Comment by a-ilango** [comment]
Good catch! Thanks. I have updated the patch to remove them if no hints are provided.

**Comment by shefty** [comment]
Are any of these needed for 1.6?

**Comment by a-ilango** [comment]
> Are any of these needed for 1.6?

This is not critical. We should wait until after 1.6 to merge.

**Pull request merged**